src/fairseq2/nn/fsdp.py

```diff
@@ -89,10 +90,19 @@ def to_fsdp(
         If ``True``, the gradients will be reduced in full precision. Only
         relevant if ``mixed_precision_dtype`` is not ``None``.
     """
+    process_group: ProcessGroup | tuple[ProcessGroup, ProcessGroup] | None = None
+
     if local_world_size is not None:
-        if local_world_size == 0:
+        if local_world_size < 1:
             raise ValueError(
-                f"`local_world_size` must be greater than 0, but is {local_world_size} instead."
+                f"`local_world_size` must be greater than 1, but is {local_world_size} instead."
+            )
+
+        if local_world_size == 1:
+            raise ValueError(
+                f"`local_world_size` must be greater than 1, but is {local_world_size} instead. "
+                "This hybrid configuration would force FSDP to switch to use `NO_SHARD`, "
```

Comment:
Nit. As a convention, we do not line-break individual sentences of an error and log message within the code. This makes it possible for users to "grep" any part of the sentence (e.g. if they see it in the log file and want to figure out where it originates in the code base). This was mostly inspired by Linux kernel coding conventions.

Reply:
Ah I actually was wondering about that, thank you for the explanation, implemented this.